apps/desktop/src/main/native-messaging.main.ts

```diff
@@ -184,23 +197,61 @@ export class NativeMessagingMain {
         break;
       }
       case "linux": {
+        // Because on linux, th path inside the sandbox is different, and we want to support:
+        // Flatpak App, Unsandboxed App, Flatpak Browser, Unsandboxed Browser, Snap App, Unsandboxed App
+        // and any combination of the above, we copy the binary to the applications native-messaging-hosts path
+        // so that a canonical path to put in the manifest can be used.
+
+        // Unsandboxed browser
         for (const [key, value] of Object.entries(this.getLinuxNMHS())) {
           if (existsSync(value)) {
+            let nhmsPath = path.join(value, "NativeMessagingHosts");
+            if (key === "Firefox") {
+              nhmsPath = path.join(value, "native-messaging-hosts");
+            }
+            const browserBinaryPath = path.join(nhmsPath, ".bitwarden_desktop_proxy");
+
+            await fs.mkdir(nhmsPath, { recursive: true });
+            await fs.copyFile(binaryPath, browserBinaryPath);
+            this.logService.info(`Copied ${binaryPath} to ${browserBinaryPath}`);
+
             if (key === "Firefox") {
               await this.writeManifest(
-                path.join(value, "native-messaging-hosts", "com.8bit.bitwarden.json"),
-                firefoxJson,
+                path.join(nhmsPath, "com.8bit.bitwarden.json"),
+                await this.generateFirefoxJson(browserBinaryPath),
               );
             } else {
               await this.writeManifest(
-                path.join(value, "NativeMessagingHosts", "com.8bit.bitwarden.json"),
-                chromeJson,
+                path.join(nhmsPath, "com.8bit.bitwarden.json"),
+                await this.generateChromeJson(browserBinaryPath),
               );
             }
           } else {
             this.logService.warning(`${key} not found, skipping.`);
           }
         }
+
+        for (const [key, value] of Object.entries(this.getFlatpakNMHS())) {
+          if (existsSync(value)) {
+            const sandboxedProxyBinaryPath = path.join(value, "bitwarden_desktop_proxy");
+            await fs.copyFile(binaryPath, path.join(value, ".bitwarden_desktop_proxy"));
+            this.logService.info(
+              `Copied ${sandboxedProxyBinaryPath} to ${path.join(value, "bitwarden_desktop_proxy")}`,
+            );
```

Comment:
Is this logic correct? We're copying the file to `.bitwarden_desktop_proxy` but then creating a manifest for `bitwarden_desktop_proxy` without the dot. If I understand it correctly it should probably be something more like:
```suggestion
            const sandboxedProxyBinaryPath = path.join(value, ".bitwarden_desktop_proxy");
            await fs.copyFile(binaryPath, sandboxedProxyBinaryPath);
            this.logService.info(
              `Copied ${binaryPath} to ${sandboxedProxyBinaryPath}`,
            );
```